### PR TITLE
add e2e test for http3

### LINKED ISSUE #2423: E2E for HTTP3
*Description*:
>Describe the issue.

Relates to https://github.com/envoyproxy/gateway/pull/2111

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by zesiar0**
**What type of PR is this?**
* feat: add e2e test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #2423

### CODE REVIEW

File: test/e2e/tests/http3.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+			gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+			expectedResponse := http.ExpectedResponse{
+				Request: http.Request{
+					Path: "/get",
+					Headers: map[string]string{
+						"Host": "www.example.com",
+					},
+				},
+				Response: http.Response{
+					StatusCode: 200,
```
Comment (arkodg) [inline test/e2e/tests/http3.go:54]: any other h3 response headers that can be used in validation ?

File: test/e2e/base/manifests.yaml [hunk 2]
```diff
@@ -521,3 +521,138 @@ spec:
...
+        name: http3-secret
+        namespace: gateway-conformance-http3
+  - name: http
+    protocol: HTTP
+    port: 80
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: http3-backend
```
Comment (shawnh2) [inline test/e2e/base/manifests.yaml:608]: according to [ci error trace](https://github.com/envoyproxy/gateway/actions/runs/7606054472/job/20711517028?pr=2468#step:6:1071), the namespace of SA should be set

File: test/e2e/tests/http3.go (hunk 1 above)
Comment (zesiar0) [reply test/e2e/tests/http3.go:54]: ok, I will add more http3 response headers later

File: test/e2e/base/manifests.yaml (hunk 2 above)
Comment (zesiar0) [reply test/e2e/base/manifests.yaml:608]: Thanks

**Pull request closed without merging**